airflow/decorators/base.py

```diff
@@ -255,6 +255,21 @@ def __attrs_post_init__(self):
         self.kwargs.setdefault('task_id', self.function.__name__)
 
     def __call__(self, *args, **kwargs) -> XComArg:
+
+        task_id = kwargs.pop('task_id', None)
+        owner = kwargs.pop('owner', None)
+        email = kwargs.pop('email', None)
+        pool = kwargs.pop('pool', None)
+        priority_weight = kwargs.pop('priority_weight', None)
+        new_params = {
+            'task_id': task_id,
+            'owner': owner,
+            'email': email,
+            'pool': pool,
+            'priority_weight': priority_weight,
+        }
+        self.kwargs.update({k: v for k, v in new_params.items() if v is not None})
+
```

Comment:
Hi @ashb, this is one way I have found that we could allow more `Baseoperator` arguments. Let me know if this looks good and I'll expand on it. Thanks